data/src/main/java/com/linkedin/data/schema/EnumDataSchema.java

```diff
@@ -170,6 +172,25 @@ public boolean contains(String symbol)
     return _symbolToIndexMap.containsKey(symbol);
   }
 
+  /**
+   * Returns properties for the given symbol.
+   * @param symbol to get properties for.
+   * @return properties for the symbol (empty map if no properties defined). null for invalid symbols.
+   */
+  public Map<String, Object> getSymbolProperties(String symbol)
+  {
+    if (!_symbolToIndexMap.containsKey(symbol))
+    {
+      return null;
+    }
+    Object prop = getProperties().get(SYMBOL_PROPERTIES_KEY);
+    if(prop instanceof DataMap)
+    {
+      return ((DataMap) prop).getDataMap(symbol);
```

Comment:
Does this “getDataMap" require assumption that the symbolPropertiesMap always well-formatted?  Do we do any check elsewhere? Is this guaranteed?

Reply:
This is guranteed by the schema parser.